### PR TITLE
Terra phoenix airdrop : Updated airdrop contract to adapt to cw-1.0

### PR DESCRIPTION
**Pull request by alienHunterOnMars**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ cargo build
 docker run --rm -v "$(pwd)":/code \
   --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
   --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
-  cosmwasm/rust-optimizer:0.11.3
+  cosmwasm/workspace-optimizer:0.12.3
```
Comment (ShadoySV) [inline README.md:53]: Let's use 0.12.6

**Review by ShadoySV — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 [workspace]
 members = [
-    "contracts/airdrop", 
+    # "contracts/airdrop", 
     "contracts/simple_airdrop", 
-    "contracts/auction", 
-    "contracts/lockdrop", 
+    # "contracts/auction",
```
Comment (epanchee) [inline Cargo.toml:5]: Cargo.lock is outdated. It needs to run `cargo update`

File: contracts/simple_airdrop/Cargo.toml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 [package]
-name = "simple-astroport-airdrop"
+name = "phoenix-astroport-airdrop"
```
Comment (epanchee) [inline contracts/simple_airdrop/Cargo.toml:2]: I'm not sure that we have to rename crate's name

File: contracts/simple_airdrop/Cargo.toml [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
 version = "1.0.0"
 authors = ["_astromartian"]
 edition = "2018"
```
Comment (epanchee) [inline contracts/simple_airdrop/Cargo.toml:5]: Update edition to 2021 as well

File: packages/astroport_periphery/Cargo.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "astroport-periphery"
-version = "1.1.0"
+version = "1.0.0"
 authors = ["_astromartian"]
 edition = "2018"
```
Comment (epanchee) [inline packages/astroport_periphery/Cargo.toml:5]: Update edition to 2021 as well

File: packages/astroport_periphery/Cargo.toml [hunk 6]
```diff
@@ -24,15 +24,13 @@ overflow-checks = true
...
-cw20 = { version = "0.8" }
-
-terra-cosmwasm = { version = "2.2.0" }
-cosmwasm-std = { version = "0.16.0", features = ["iterator"] }
-cosmwasm-bignumber = "2.2.0"
-terraswap = "2.4.0"
-
+cw20 = { version = "0.13.2" }
+# terra-cosmwasm = { version = "2.2.0" }
+cosmwasm-std = { version = "1.0.0" }
```
Comment (epanchee) [inline packages/astroport_periphery/Cargo.toml:29]:
```suggestion
cosmwasm-std = "1.0"
```

Use 1.0 so cargo will be able to update patch versions

File: packages/astroport_periphery/src/lib.rs [hunk 7]
```diff
@@ -1,5 +1,5 @@
-pub mod airdrop;
-pub mod auction;
+// pub mod airdrop;
```
Comment (epanchee) [inline packages/astroport_periphery/src/lib.rs:1]: There is no need to comment these libs as you already commented these contracts in the top-level Cargo.toml

File: contracts/simple_airdrop/src/contract.rs [hunk 8]
```diff
@@ -131,7 +131,6 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
     match msg {
         QueryMsg::Config {} => to_binary(&query_config(deps)?),
         QueryMsg::State {} => to_binary(&query_state(deps)?),
-        QueryMsg::HasUserClaimed { address } => to_binary(&query_user_claimed(deps, address)?),
```
Comment (epanchee) [inline contracts/simple_airdrop/src/contract.rs:134]: Why?

**Review by epanchee — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (alienHunterOnMars) [reply README.md]: resolved

File: Cargo.toml (hunk 2 above)
Comment (alienHunterOnMars) [reply Cargo.toml:5]: i updated locally but hadn't committed, doing now

File: contracts/simple_airdrop/Cargo.toml (hunk 3 above)
Comment (alienHunterOnMars) [reply contracts/simple_airdrop/Cargo.toml:2]: simple airdrop didn't look like appropriate naming, lmk if i should change it

File: packages/astroport_periphery/Cargo.toml (hunk 5 above)
Comment (alienHunterOnMars) [reply packages/astroport_periphery/Cargo.toml]: resolved

File: packages/astroport_periphery/src/lib.rs (hunk 7 above)
Comment (alienHunterOnMars) [reply packages/astroport_periphery/src/lib.rs]: ok resolved

File: contracts/simple_airdrop/src/contract.rs (hunk 8 above)
Comment (alienHunterOnMars) [reply contracts/simple_airdrop/src/contract.rs:134]: i updated the logic a bit so we can update the merkle roots if need be so a user can claim airdrop multiple times, in which case this query doesn't make sense

**Review by maksk7 — APPROVED** [review]

File: contracts/simple_airdrop/Cargo.toml (hunk 3 above)
Comment (epanchee) [reply contracts/simple_airdrop/Cargo.toml:2]: Ok, makes sense

File: contracts/simple_airdrop/src/contract.rs (hunk 8 above)
Comment (epanchee) [reply contracts/simple_airdrop/src/contract.rs:134]: gotcha

**Comment by epanchee** [comment]
@alienHunterOnMars fix schema generator and integration tests

**Review by P-Yevhenii — APPROVED** [review]

**Review by epanchee — APPROVED** [review]

**Review by ShadoySV — APPROVED** [review]

**Pull request merged**